pkg/machine/wsl/machine.go

```diff
@@ -0,0 +1,1093 @@
+	if err != nil {
+		return errors.Wrap(err, "could not create root authorized keys on guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", key+"\n", "-d", dist, "sh", "-c",
+		"mkdir -p /home/user/.ssh; cat >> /home/user/.ssh/authorized_keys; chown -R user:user /home/user/.ssh; chmod 600 /home/user/.ssh/authorized_keys")
+	if err != nil {
+		return errors.Wrap(err, "could not create 'user' authorized keys on guest OS")
+	}
+
+	return nil
+}
+
+func configureSystem(v *MachineVM, dist string) error {
+	err := runCmdPassThrough("wsl", "-d", dist, "sh", "-c", fmt.Sprintf(appendPort, v.Port, v.Port))
+	if err != nil {
+		return errors.Wrap(err, "could not configure SSH port for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", configServices, "-d", dist, "sh")
+	if err != nil {
+		return errors.Wrap(err, "could not configure systemd settomgs for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", sudoers, "-d", dist, "sh", "-c", "cat >> /etc/sudoers")
+	if err != nil {
+		return errors.Wrap(err, "could not add wheel to sudoers")
+	}
+
+	err = pipeCmdPassThrough("wsl", overrideSysusers, "-d", dist, "sh", "-c",
+		"cat > /etc/systemd/system/systemd-sysusers.service.d/override.conf")
+	if err != nil {
+		return errors.Wrap(err, "could not generate systemd-sysusers override for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", lingerService, "-d", dist, "sh", "-c",
+		"cat > /home/user/.config/systemd/user/linger-example.service")
+	if err != nil {
+		return errors.Wrap(err, "could not generate linger service for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", lingerSetup, "-d", dist, "sh")
+	if err != nil {
+		return errors.Wrap(err, "could not configure systemd settomgs for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", containersConf, "-d", dist, "sh", "-c", "cat > /etc/containers.conf")
+	if err != nil {
+		return errors.Wrap(err, "could not create containers.conf for guest OS")
+	}
+
+	return nil
+}
+
+func installScripts(dist string) error {
+	err := pipeCmdPassThrough("wsl", enterns, "-d", dist, "sh", "-c", "cat > /usr/local/bin/enterns; chmod 755 /usr/local/bin/enterns")
+	if err != nil {
+		return errors.Wrap(err, "could not create enterns script for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", profile, "-d", dist, "sh", "-c", "cat > /etc/profile.d/enterns.sh")
+	if err != nil {
+		return errors.Wrap(err, "could not create motd profile script for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", wslmotd, "-d", dist, "sh", "-c", "cat > /etc/wslmotd")
+	if err != nil {
+		return errors.Wrap(err, "could not create a WSL MOTD for guest OS")
+	}
+
+	err = pipeCmdPassThrough("wsl", bootstrap, "-d", dist, "sh", "-c", "cat > /root/bootstrap; chmod 755 /root/bootstrap")
+	if err != nil {
+		return errors.Wrap(err, "could not create bootstrap script for guest OS")
+	}
+
+	return nil
+}
+
+func checkAndInstallWSL(opts machine.InitOptions) (bool, error) {
+	if !isWSLInstalled() {
```

Comment:
Could be rewritten to `if isWLSInstalled() { return true, nil}` and drop the scope.